packages/hardhat/contracts/strategy/CrabStrategyV2.sol

```diff
@@ -722,188 +572,156 @@ contract CrabStrategyV2 is StrategyBase, StrategyFlashSwap, ReentrancyGuard, Own
+            IWETH9(weth).deposit{value: _order.managerAmount}();
+            // if last param is false, transfer happens again
+            priceAtLastHedge = _order.managerAmount.mul(1e18).div(_order.traderAmount);
         }
 
-        emit ExecuteBuyAuction(_seller, _wSqueethToBuy, _ethToSell, _isHedgingOnUniswap);
+        IERC20(_order.managerToken).transfer(_order.trader, _order.managerAmount);
+
+        emit HedgeOTC(
+            _order.trader, // market maker
+            _order.managerAmount, // token out
+            _order.traderAmount, // token in
+            sellerPrice
+        );
     }
 
     /**
-     * @notice determine auction direction, price, and ensure auction hasn't switched directions
-     * @param _auctionTriggerTime auction starting time
-     * @return auction type
-     * @return WSqueeth amount to sell or buy
-     * @return ETH to sell/buy
-     * @return auction WSqueeth/ETH price
+     * @dev hedge function to reduce delta using an array of signed orders
+     * @param managerSellAmount quantity the manager wants to sell
+     * @param managerBuyPrice the price at which the manager is buying
+     * @param _orders an array of signed order to swap tokens
      */
-    function _startAuction(uint256 _auctionTriggerTime)
-        internal
-        returns (
-            bool,
-            uint256,
-            uint256,
-            uint256
-        )
-    {
-        (uint256 strategyDebt, uint256 ethDelta) = _syncStrategyState();
-        uint256 currentWSqueethPrice = IOracle(oracle).getTwap(
-            ethWSqueethPool,
-            wPowerPerp,
-            weth,
-            hedgingTwapPeriod,
-            true
-        );
-        uint256 feeAdjustment = _calcFeeAdjustment();
-        (bool isSellingAuction, ) = _checkAuctionType(strategyDebt, ethDelta, currentWSqueethPrice, feeAdjustment);
-        uint256 auctionWSqueethEthPrice = _getAuctionPrice(_auctionTriggerTime, currentWSqueethPrice, isSellingAuction);
-        (bool isStillSellingAuction, uint256 wSqueethToAuction) = _checkAuctionType(
-            strategyDebt,
-            ethDelta,
-            auctionWSqueethEthPrice,
-            feeAdjustment
-        );
+    function hedgeOTC(
+        uint256 managerSellAmount,
+        uint256 managerBuyPrice,
+        Order[] memory _orders
+    ) external onlyOwner {
+        require(managerBuyPrice > 0, "Manager Price should be greater than 0");
+        require(_isTimeHedge() || _isPriceHedge(), "Time or Price is not within range");
+        _checkOTCPrice(managerBuyPrice, _orders[0].managerToken);
 
-        require(isSellingAuction == isStillSellingAuction, "auction direction changed");
+        timeAtLastHedge = block.timestamp;
 
-        uint256 ethProceeds = wSqueethToAuction.wmul(auctionWSqueethEthPrice);
+        uint256 remainingAmount = managerSellAmount;
+        uint256 prevPrice = 0;
+        uint256 currentPrice = 0;
+        bytes memory tradePair;
+        bytes memory prevTradePair;
```

Comment:
I think can save some gas by optimizing this trade pair check.

Can compare each tradePair to the order[0].tradePair

``` bytes memory firstTradePair = abi.encode(_orders[0].traderToken, _orders[0].managerToken);
```

then, just requiring that ``` keccak256(tradePair) == keccak256(firstTradePair), ````

Additionally, any reason to not sure the keccak256 instead of the abi bytes? ie

``` bytes memory firstTradePair = keccak256(abi.encode(_orders[0].traderToken, _orders[0].managerToken)); ```

That way you only have to hash the firstTradePair 1x.